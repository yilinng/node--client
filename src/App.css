.container {
  margin: auto;
  text-align: center;
  height: 100%;
}

.haveLoading {
  opacity: 0.5;
}

/* login and signup*/
form {
  display: block;
}

label {
  display: inline-block;
  width: 100px;
  margin: auto;
}

form input {
  height: 40px;
  width: 400px;
  border: none;
  border-bottom: 2px #101113 solid;
  background: none;
  font-size: 18px;
  color: #787a80;
}

input:focus {
  outline: none;
}

form button {
  margin-top: 20px;
  width: 150px;
  height: 50px;
  font-size: 25px;
  border: none;
  background-color: #0000CD;
  color: #EE82EE;
  font-weight: 500;
  border-radius: 10rem;
  cursor: pointer;
}

.passwordConfirm  {
  margin-top: 30px;
}

/* Dashboard */
.cardBody {
  width: auto;
  height: auto;
  display: block;
}

.nullMessage{
  width: 75%;
  height: 40px;
}

.boardMessage {
  display: block;
  margin: auto;
  width: 75%;
  height: 40px;
  background-color: #2E8B57;
  border-radius: 15px;
}

.boardMessage .close {
  float: right;
}

.boardMessage .close:hover {
  color: #eee;
}

.boardMessage .msg {
  font-size: 20px;
  font-weight: 500;
  margin: auto;
  vertical-align: bottom;
}



.textwelcome {
  margin: 10px auto;
  font-size: 25px;
}

.textwelcome .name{
  color:olivedrab;
  font-weight: 600;
}

.link a{
  display: block;
  margin: 10px auto;
  font-size: 20px;
  font-weight: 500;
}

.link a:hover{
  color: yellowgreen;
}

.logout button {
  width: 75px;
  height: 30px;
  font-size: 17px;
  border: none;
  border-radius: 10px;
  background-color: #3498db;
  color: #eee;
}

.logout button:hover{
  background-color: #1B4E70;
  color: #eee;
}

/* todo board */
header{
  position: fixed;
  height: 80px;
  width: 100%;
  background-color: #fafafa;
  z-index: 10;
  display: flex;
  justify-content: space-between;
}

header span {
  margin: 1.5%;
}

header .title {
  font-size: 20px;
  font-weight: 500;
}

header svg {
  height: 30px;
  width: 30px;
  cursor: pointer;
}

.todo {
  display: flex;
  height: 100vh;
}

.sideBarTodo{
  position: relative;
  height: auto;
  background-color: #f1f1f1;
  display: none;
  margin-top: 4.5%;
}

 .active{
  width: 20%;
  display: block;
}

.sideBarTodo .list h3 {
  text-align: center;
  border-bottom: azure 1px solid;
}

.sideBarTodo .list .li{
 display: flex;
 justify-content: space-around; 
 text-align: center;
 margin: 5px;
 border-bottom: azure 1px solid;
 cursor: pointer;
}


.sideBarTodo .list .li:hover{
  background-color: cadetblue;
  color: #eee;
 }

 .sideBarTodo .list .li svg{
  width: 20px;
  height: 20px;
  border-radius: 5px;
 }

 .sideBarTodo .list .li .minMenu:active, .sideBarTodo .list .li .minMenu:hover{
  background-color: #787a80;
  color: white;
 }

 .sideBarTodo .list .action {
   position: absolute;
   z-index: 10;
   background-color: #fafafa;
   display: flex;
   flex-direction: column;
   width: 150px;
   height: 120px;
   text-align: center;
   border:#eee 1px solid;
 }

 .sideBarTodo .list .action span {
   width: 90%;
   height: auto;
   margin: auto;
   cursor: pointer;
   color: black;
 }

 .sideBarTodo .list .action span:hover, .sideBarTodo .list .action span:active{
  background-color: #787a80;
  color: #eee;
  font-weight: 500;
}


.sideBarTodo .list .action svg{
  width: 20px;
  height: 20px;
  vertical-align: bottom;
}

.sideBarTodo .list .addOne {
  cursor: pointer;
  width: 90%;
}

.sideBarTodo .list .addOne:hover {
  cursor: pointer; 
  background-color: #787a80;
  color: #eee;
}

.whiteBoard {
  margin: 7% 15%;
  width: 65%;
 display: block;
}

.whiteBoard .updateBtn {
  font-size: 20px;
  font-weight: 500;
  background-color: forestgreen;
  border: none;
  color: white;
  cursor: pointer;
  border-radius: 10px;
  padding: 5px;
}

.whiteBoard .tips{
  display: block;
  color: #0e0950;
  font-size: 18px;
}

.whiteBoard .main{
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  height: auto;
}

.whiteBoard .main .title{
  margin-top: 2%;
}


.whiteBoard .main .title textarea{
  font-size: 30px;
  resize: none;
  border: none;
  font-weight: 700;
  border-bottom: #eee 1px solid;
  width: 600px;
  min-height: 50px;
  white-space: pre-wrap;
}

.whiteBoard .main .contextList{
  margin: 20px auto;
}

.whiteBoard .main .contextList .context{
  display: flex;
  flex-direction: row;
  margin: 30px auto;
}



.whiteBoard .main .contextList .context .contextText{
  word-break: break-word;
  box-sizing: border-box;
}

.whiteBoard .main .contextList .context .contextText > textarea {
  height: 100%;
  border: none;
  border-bottom: #eee 1px solid;
  font-size: 16px;
  width: 600px;
  white-space: pre-wrap;
  resize: none;
}


.whiteBoard .main .contextList .context .contextText > div {
  padding-bottom: 1.5em; /* A bit more than one additional line of text. */ 
  visibility: hidden;
}


.whiteBoard .main .contextList .context .contextMenu{
  width: 20px;
  height: 20px;
}

.whiteBoard .main .contextList .context .contextActionList{
  position: absolute;
  display: flex;
  flex-direction: column;
  background-color: #eee;
  width: 66px;
  height: 80px;
  padding: 2px 5px;
  border: 1px solid	#C0C0C0;
}

.whiteBoard .main .contextList .context .contextActionList div {
  cursor: pointer;
  display: flex;
  flex-direction: row;
}

.whiteBoard .main .contextList .context .contextActionList div:hover, .whiteBoard .main .contextList .context .contextActionList div:active{
  background-color: darkblue;
  color: white;
}

.whiteBoard .main .contextList .context svg {
  width: 20px;
  height: 20px;
  cursor: pointer;
}
.whiteBoard .main .contextList .context svg:hover, .whiteBoard .main .contextList .context svg:active{
  background-color: #787a80;
  color: #eee;
}


.loader {
  border: 16px solid #f3f3f3;
  border-radius: 50%;
  border-top: 16px solid #3498db;
  width: 100px;
  height: 100px;
  -webkit-animation: spin 1.1s linear infinite; /* Safari */
  animation: spin 1.1s linear infinite;
  z-index: 100;
  position: absolute;
  margin: 10% auto;
  left: 0;
  right: 0;
}


/* Safari */
@-webkit-keyframes spin {
  0% { -webkit-transform: rotate(0deg); }
  100% { -webkit-transform: rotate(360deg); }
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}

@media screen and (max-width: 670px){

  header {
    height: 40px;
  }

  header span {
    margin: 10px;
  }

  header svg {
    height: 20px;
    width: 20px;
  }


  .sideBarTodo{
    position: absolute;
    z-index: 5;
    margin-top: 40px;
  }
  
   .active{
    width: 75%;
    display: block;
  }

  .whiteBoard {
    margin: 60px auto;
    display: block;
  }

  .whiteBoard .main .title{
    margin-top: 5%;
  }

  .whiteBoard .main .title textarea{
    font-size: 25px;
    width: 243px;
    min-height: 45px;
  }

  .whiteBoard .main .contextList .context .contextText > textarea {
    font-size: 16px;
    width: 243px;
    min-height: 32px;
  }

  form input {
    width: 170px;
  }
  
}

@media screen and (max-width: 1024px) and (min-width: 768px){

  header{
    height: 55px;
  }

  .sideBarTodo{
    position: fixed;
    height: 100%;
  }

  .active{
    width: 40%;
  }

  .whiteBoard {
    margin: 9% auto;
  }

  form input {
    width: 250px;
  }

}