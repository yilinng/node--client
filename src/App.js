import './App.css';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { AuthProvider } from './contexts/AuthContext';
import Signup from './components/Signup';
import Dashboard from './components/Dashboard';
import Login from './components/Login';
import PrivateRoute from './components/PrivateRoute';
import ForgotPassword from './components/ForgotPassword';
import UpdateProfile from './components/UpdateProfile';
import Todo from './components/Todo';

function App() {
 
  return (
    <div className="App">
        <Router>
          <AuthProvider>
            <Switch>
              <PrivateRoute exact path="/" component={Dashboard}/>
              <PrivateRoute path ="/update-profile" component={UpdateProfile}/>
              <PrivateRoute path ="/todos" component={Todo}/>
              <Route path="/signup" component={Signup}/>
              <Route path="/login" component={Login}/>
              <Route path="/forgot-password" component={ForgotPassword}/>
            </Switch>
          </AuthProvider>
        </Router>
    </div>
  );
}

export default App;