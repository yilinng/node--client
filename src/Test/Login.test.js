import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
//import Login from '../components/Login'
import App from '../App'
import TestRenderer from 'react-test-renderer';
describe('Login tests', () => {

  it('should contains button with button toBeEnabled', () => {
    render(<App />);
      userEvent.type(screen.getByRole("textbox", {id: "email"}), "abc@email.com");
      userEvent.type(screen.getByRole("textbox", {id: "password"}), "abc@email.com");
      const button = screen.getByRole("button", {name: "log In"});
      expect(button).toBeEnabled()
  });

  it('test to match snapshot of component', () => {
      const loginTree = TestRenderer.create(<App/>).toJSON()
      expect(loginTree).toMatchSnapshot();
    });
});