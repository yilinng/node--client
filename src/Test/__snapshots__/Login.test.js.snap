// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login tests test to match snapshot of component 1`] = `
<div
  className="App"
/>
`;
